libs/vkd3d-shader/spirv.c

```diff
@@ -5069,6 +5080,24 @@ static const struct vkd3d_shader_global_binding *vkd3d_dxbc_compiler_get_global_
             vkd3d_spirv_enable_capability(builder, SpvCapabilityUniformBufferArrayNonUniformIndexingEXT);
         }
     }
+    else if (flags & VKD3D_SHADER_GLOBAL_BINDING_RAW_SSBO)
+    {
+        uint32_t array_type_id = vkd3d_spirv_build_op_type_runtime_array(builder,
+                vkd3d_spirv_get_type_id(builder, VKD3D_TYPE_UINT, 1));
+        vkd3d_spirv_build_op_decorate1(builder, array_type_id, SpvDecorationArrayStride, 4);
+
+        type_id = vkd3d_spirv_build_op_type_struct(builder, &array_type_id, 1);
+        vkd3d_spirv_build_op_decorate(builder, type_id, SpvDecorationBufferBlock, NULL, 0);
+        vkd3d_spirv_build_op_member_decorate1(builder, type_id, 0, SpvDecorationOffset, 0);
+
+        if (data_type == VKD3D_DATA_RESOURCE)
```

Comment:
I don't think we need to decorate the type itself when we decorate the variable declaration.